src/tigerbeetle/repl.zig

```diff
@@ -0,0 +1,810 @@
+                }
+
+                if (i > 0) {
+                    print(",\n", .{});
+                }
+
+                if (comptime std.mem.eql(u8, s_field.name, "flags")) {
+                    print("  \"" ++ s_field.name ++ "\": [", .{});
+                    var needs_comma = false;
+
+                    inline for (@typeInfo(s_field.field_type).Struct.fields) |flag_field| {
+                        if (comptime !std.mem.eql(u8, flag_field.name, "padding")) {
+                            if (@field(@field(object, "flags"), flag_field.name)) {
+                                if (needs_comma) {
+                                    print(",", .{});
+                                    needs_comma = false;
+                                }
+
+                                print("\"" ++ flag_field.name ++ "\"", .{});
+                                needs_comma = true;
+                            }
+                        }
+                    }
+
+                    print("]", .{});
+                } else {
+                    print("  \"" ++ s_field.name ++ "\": \"{}\"", .{@field(object, s_field.name)});
+                }
+            }
+            print("\n}}\n", .{});
+        }
+
+        fn display_transfer_result_errors(errors: []align(1) const tb.CreateTransfersResult) void {
+            for (errors) |reason| {
+                print(
+                    "Failed to create transfer ({}): {any}.\n",
+                    .{ reason.index, reason.result },
+                );
+            }
+        }
+
+        fn send_complete(
+            user_data: u128,
+            operation: StateMachine.Operation,
+            result: []const u8,
+        ) void {
+            const context = @intToPtr(*Context, @intCast(u64, user_data));
+            assert(context.request_done == false);
+            context.debug("Operation completed: {}.\n", .{operation});
+
+            defer {
+                context.request_done = true;
+                context.client.unref(context.message.?);
+                context.message = null;
+
+                if (!context.interactive) {
+                    context.event_loop_done = true;
+                }
+            }
+
+            switch (operation) {
+                .create_accounts => {
+                    const create_account_results = std.mem.bytesAsSlice(
+                        tb.CreateAccountsResult,
+                        result,
+                    );
+
+                    if (create_account_results.len > 0) {
+                        display_account_result_errors(create_account_results);
+                    }
+                },
+                .lookup_accounts => {
+                    const lookup_account_results = std.mem.bytesAsSlice(
+                        tb.Account,
+                        result,
+                    );
+
+                    if (lookup_account_results.len == 0) {
+                        context.fail("No such account exists.\n", .{});
```

Comment:
It can be any non-zero number of ids passed in. This was just a nice message if none of them are returned. (Understanding that there's the separate [unacknowledged] case where some but not all are returned.)

Reply:
I've adjusted the message slightly.